vowpalwabbit/core/src/io_buf.cc

```diff
@@ -3,11 +3,33 @@
 // license as described in the file LICENSE.
 #include "vw/core/io_buf.h"
 
-size_t VW::io_buf::buf_read(char*& pointer, size_t n)
+#if false  // AUDIT IO BUFFER ALIGNMENTS
+#  define __AUDIT_VW_IO_BUF(operation, target_align)                                                                \
+    std::cerr.flush();                                                                                              \
+    std::cerr << std::endl                                                                                          \
+              << "+++ io_buf " #operation ": @" << std::hex << reinterpret_cast<size_t>(_head) << std::dec << " % " \
+              << target_align.align << " = " << reinterpret_cast<size_t>(_head) % target_align.align << " vs "      \
+              << target_align.offset << ")" << std::endl;
+#else
+#  define __AUDIT_VW_IO_BUF(operation, target_align)
+#endif
+
+size_t VW::io_buf::buf_read(char*& pointer, size_t n, desired_align align)
```

Comment:
Adding some explanatory comments (and a bit of backstory) here would be very useful.  This was a super interesting.